PL2/PL2-shared/src/main/java/nl/tudelft/pl2016gr2/model/OriginalGraph.java

```diff
@@ -6,17 +6,23 @@
 public class OriginalGraph implements GraphInterface {
 
   private HashMap<Integer, Node> nodes;
+  private HashMap<Integer, AbstractNode> abstractNodes;
```

Comment:
I would like to add that if the `OriginalGraph` uses `Node`, it should override, using covariance, the return types of the `GraphInterface`. The interface should not specify two types of nodes, and its implementations should not offer two types of the same inheritance.

If OriginalGraph doesn't necessarily use `Node`, but you want an OriginalGraph that does, simply extend it.